lib/services/elasticsearch.js

```diff
@@ -919,162 +949,123 @@ class ElasticSearch extends Service {
   }
 
   /**
-   * Add a mapping definition to a specific type
+   * Retrieve the complete list of existing collections in the current index
    *
-   * @param {Request} request
-   * @return {Promise}
-   */
-  updateMapping(request) {
-    const esRequest = initESRequest(request);
-
-    return this.esWrapper.getMapping(esRequest, true)
-      .then(mappings => {
-        const collectionMapping =
-          mappings[esRequest.index].mappings[esRequest.type];
-        const requestBody = request.input.body || {};
-
-        this._checkMapping(requestBody);
-
-        esRequest.body = this._mergeDefaultMapping(
-          esRequest.index,
-          collectionMapping,
-          requestBody
-        );
-
-        return this.client.indices.putMapping(esRequest)
-          .catch(error => Bluebird.reject(
-            this.esWrapper.formatESError(error))
-          );
-      });
-  }
-
-  /**
-   * Retrieve mapping definition for index/type
+   * @param {string} index - Index name
    *
-   * @param {Request} request
-   * @return {Promise}
+   * @return {Promise} { collections }
    */
-  getMapping(request, includeKuzzleMeta = false) {
-    const esRequest = initESRequest(request);
+  listCollections (index) {
+    return this.client.cat.indices({ format: 'json' })
+      .then(({ body }) => {
+        const
+          esIndexes = body.map(({ index }) => index),
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') "index" hides or potentially hides a variable declared in an outer scope at line 958. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AVariableShadowing)